module/VuFind/src/VuFind/Controller/Plugin/Holds.php

```diff
@@ -268,4 +300,42 @@ public function validateDates($startDate, $requiredBy, $enabledFormFields)
         $result['errors'] = $errors;
         return $result;
     }
+
+    /**
+     * Check if the user-provided "frozen until" date is valid.
+     *
+     * Returns validated date and/or an array of validation errors if there are
+     * problems.
+     *
+     * @param string $frozenUntil     User-specified "frozen until" date
+     * @param array  $extraHoldFields Hold form fields enabled by
+     * configuration/driver
+     *
+     * @return array
+     */
+    public function validateFrozenUntil(?string $frozenUntil, array $extraHoldFields
+    ): array {
+        $result = [
+            'frozenUntilTS' => null,
+            'errors' => [],
+        ];
+        if (!in_array('frozenUntil', $extraHoldFields) || empty($frozenUntil)) {
+            return $result;
+        }
+
+        $errors = [];
```

Comment:
Since you initialize $result['errors'] above, would it be easier to just add messages directly to that instead of introducing another $errors variable and reassigning it later?

Reply:
Simplified.